### PR TITLE
Rule Correction for deprecated Rules etc.

### PR DESCRIPTION
**Pull request by twitnic**
corrected old rules to newer sdk rules
(Source: htdocs/sites/all/vendor/facebook/facebook-instant-articles-sdk-php/tests/Facebook/InstantArticles/Transformer/custom-html-rules.json )

Now it is possible to have a Vimeo-iFrame Link in the Content.

### CODE REVIEW

**Comment by twitnic** [comment]
Now it is possible to have a Vimeo-iFrame Link in the Content.

**Comment by m4olivei** [comment]
@twitnic Is there a corresponding issue in the drupal.org issue queue for this?

File: fb_instant_articles.module [hunk 1]
```diff
@@ -265,28 +265,18 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
       'selector' => 'blockquote',
     ),
     array(
-      'class' => 'InteractiveRule',
```
Comment (m4olivei) [inline fb_instant_articles.module:268]: Why is this rule going away?

File: fb_instant_articles.module [hunk 2]
```diff
@@ -265,28 +265,18 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
...
-          'selector' => 'dl',
-        ),
-      ),
+      'class' => 'PassThroughRule',
+      'selector' => 'blockquote p',
     ),
     array(
-      'class' => 'ImageRule',
-      'selector' => '//p[img]',
+      'class' => 'ImageInsideParagraphRule',
```
Comment (m4olivei) [inline fb_instant_articles.module]: Nice.

File: fb_instant_articles.module [hunk 3]
```diff
@@ -265,28 +265,18 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
...
+      'selector' => 'img',
       'properties' => array(
         'image.url' => array(
           'type' => 'string',
           'selector' => 'img',
           'attribute' => 'src',
         ),
-        'image.caption' => array(
-          'type' => 'element',
-          'selector' => 'img',
```
Comment (m4olivei) [inline fb_instant_articles.module]: Can we leave this and have it be `img[@alt]`?

File: fb_instant_articles.module [hunk 4]
```diff
@@ -298,21 +288,6 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
           'selector' => 'img',
           'attribute' => 'src',
         ),
-        'image.caption' => array(
```
Comment (m4olivei) [inline fb_instant_articles.module]: Why remove this?

File: fb_instant_articles.module [hunk 5]
```diff
@@ -328,6 +303,10 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
       'selector' => 'ol',
     ),
     array(
+      'class' => 'BlockquoteRule',
```
Comment (m4olivei) [inline fb_instant_articles.module]: This is already present in near the top of the ruleset, shouldn't be needed.

File: fb_instant_articles.module [hunk 6]
```diff
@@ -354,60 +333,161 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
       ),
     ),
     array(
-      'class' => 'SocialEmbedRule',
+      'class' => 'H2Rule',
+      'selector' => 'h3,h4,h5,h6',
+    ),
+    array(
+      'class' => 'InteractiveRule',
+      'selector' => 'blockquote.instagram-media',
```
Comment (m4olivei) [inline fb_instant_articles.module]: This is a rule that you should implement in a custom module, Instagram embeds are not universal.

File: fb_instant_articles.module [hunk 7]
```diff
@@ -354,60 +333,161 @@ function fb_instant_articles_fb_instant_articles_transformer_rules() {
...
+          'selector' => 'iframe',
+          'attribute' => 'width',
         ),
       ),
     ),
     array(
-      'class' => 'SocialEmbedRule',
-      'selector' => 'div.oembed',
+      'class' => 'InteractiveRule',
+      'selector' => 'table',
```
Comment (m4olivei) [inline fb_instant_articles.module]: This is a bit of a stretch, any `table`'s match `InteractiveRule`? This also feels like more of a custom rule that you implement with `hook_fb_instant_articles_transformer_rules` in your own custom module.

**Review by m4olivei — CHANGES_REQUESTED** [review]
There are some nice changes in here, but a couple things that don't really belong in the module, and a couple questions I had.

**Comment by twitnic** [comment]
Hey @m4olivei I have only added rules from https://github.com/facebook/facebook-instant-articles-sdk-php/blob/master/src/Facebook/InstantArticles/Parser/instant-articles-rules.json or corrected existing rules.

The Facebook Rules ...

**Comment by m4olivei** [comment]
@twitnic hum, the Facebook rules don't have anything matching a `table`, also, you have to consider the rules already present and their order and better integrate them.

**Comment by m4olivei** [comment]
Closing as this hasn't seen any activity in awhile. I'm no longer personally maintaining the 7.x branch anymore :grimacing: .

**Pull request closed without merging**